### ISSUE TITLE
Model.generate encoder decoder

### ISSUE THREAD

**Issue by Jim0530** [huggingface/transformers#14288]
Instead of using model.generate()
How can I get the output of the decoder by taking the encoder hidden states and a start token to loop over the decoder model

cuz recently model.generate doesn’t fit my custom model.
Are there any examples or tutorial 
Thank you very much

**Comment by NielsRogge** [comment]
Hi,

I've answered your question on the forum: https://discuss.huggingface.co/t/generate-without-using-the-generate-method/11379

Please ask this question on our [forum](https://discuss.huggingface.co/) instead of here. We like to keep Github issues for bugs or feature requests.

Thanks!

**Comment by ummehabiba378717805** [comment]
> Hi,
> 
> I've answered your question on the forum: https://discuss.huggingface.co/t/generate-without-using-the-generate-method/11379
> 
> Please ask this question on our [forum](https://discuss.huggingface.co/) instead of here. We like to keep Github issues for bugs or feature requests.
> 
> Thanks!

# from transformers import BlipForConditionalGeneration,AutoProcessor,AutoTokenizer
# from PIL import Image
# import requests
# import torch
# processor= AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
# model= BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-base")
# tokenizer = AutoTokenizer.from_pretrained("Salesforce/blip-image-captioning-base")

# url = "http://images.cocodataset.org/val2017/000000039769.jpg"
# image = Image.open(requests.get(url, stream=True).raw)
# text = "A picture of"

decoder_input_ids = [model.decoder_input_ids]
print(decoder_input_ids)
inputs.input_ids = decoder_input_ids=torch.tensor([decoder_input_ids])
predicted_ids = []
for i in range(20):
    outputs = model(**inputs)
    logits = outputs.logits[:,i,:]
    # print(outputs.logits.shape)
    predicted_id = logits.argmax(-1)
    predicted_ids.append(predicted_id.item())
    print(tokenizer.decode([predicted_id.squeeze()]))
    # add predicted id to decoder_input_ids
    # decoder_input_ids = decoder_input_ids + [predicted_id]
    inputs.input_ids = torch.cat([inputs.input_ids, predicted_id.unsqueeze(0)], dim=1)

for this code for every image i found  same logits shape [1, 5, 30524] . and also every iteration the logits are  remain same. no change in logtis for changing the inputs.input_ids.  why i face this error? and how to solve this?

**Comment by NielsRogge** [comment]
Hi, you should take the logits of the final token in the sequence, i.e. `logits = outputs.logits[:,-1,:]`.
